nix/e2e-tests/src/phases/input_compression.py

```
@@ -0,0 +1,209 @@
+            f"URL is the client-visible 'input compression not recognized'",
+        )
+
+        # No mislabel AND we genuinely overlapped the in-flight window → the
+        # invariant held under the conditions that trigger the bug → done.
+        if counts["overlapped"]:
+            log("  in-flight window overlapped; .nar.xz never served mislabeled", B)
+            return
+
+        # Missed the window (NAR chunked before any .nar.xz landed): retry clean.
+        if attempt < ATTEMPTS:
+            log(f"  in-flight window missed; clean restart + retry", Y)
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)